controller/src/main/java/ru/mail/dimaushenko/controller/SigninServlet.java

```diff
@@ -0,0 +1,95 @@
+package ru.mail.dimaushenko.controller;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import static ru.mail.dimaushenko.constants.AttributeConstants.ATTRIBUTE_NAME_ERRORS;
+import static ru.mail.dimaushenko.constants.AttributeConstants.ATTRIBUTE_NAME_ROLES;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_KEY_PASSWORD_LENGTH;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_KEY_ROLE;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_KEY_USERNAME_LENGTH;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_KEY_USERNAME_SYMBOLS;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_VALUE_PASSWORD_LENGTH;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_VALUE_ROLE;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_VALUE_USERNAME_LENGTH;
+import static ru.mail.dimaushenko.constants.ErrorsConstants.ERROR_VALUE_USERNAME_SYMBOLS;
+import static ru.mail.dimaushenko.constants.Pages.PAGE_SIGNIN;
+import static ru.mail.dimaushenko.constants.Pages.REQUEST_LOGIN;
+import static ru.mail.dimaushenko.constants.ParameterConstants.PARAMETER_PASSWORD;
+import static ru.mail.dimaushenko.constants.ParameterConstants.PARAMETER_ROLE_ID;
+import static ru.mail.dimaushenko.constants.ParameterConstants.PARAMETER_USERNAME;
+import ru.mail.dimaushenko.service.RoleService;
+import ru.mail.dimaushenko.service.UserService;
+import ru.mail.dimaushenko.service.impl.RoleServiceImpl;
+import ru.mail.dimaushenko.service.impl.UserServiceImpl;
+import ru.mail.dimaushenko.service.model.AddUserDTO;
+import ru.mail.dimaushenko.service.model.RoleDTO;
+import ru.mail.dimaushenko.utils.Validator;
+import ru.mail.dimaushenko.utils.impl.ValidatorImpl;
+
+public class SigninServlet extends ManagerServlet {
+
+    private final RoleService roleService = RoleServiceImpl.getInstance();
+    private final Validator validator = ValidatorImpl.getInstaice();
+    private final UserService userService = UserServiceImpl.getInstance();
+    private final Map<String, String> errorMessages = new HashMap<>();
```

Comment:
This is a temp variable for one request. Why do we use it as fields for all requests?